tests/taker_cli.rs

```diff
@@ -0,0 +1,255 @@
+
+        // Remove if previously existing
+        if temp_dir.exists() {
+            fs::remove_dir_all::<PathBuf>(temp_dir.clone()).unwrap();
+        }
+
+        let mut conf = Conf::default();
+
+        conf.args.push("-txindex=1"); //txindex is must, or else wallet sync won't work.
+        conf.staticdir = Some(temp_dir.join(".bitcoin"));
+
+        log::info!("bitcoind configuration: {:?}", conf.args);
+
+        let os = std::env::consts::OS;
+        let arch = std::env::consts::ARCH;
+        let key = "BITCOIND_EXE";
+        let curr_dir_path = std::env::current_dir().unwrap();
+
+        let bitcoind_path = match (os, arch) {
+            ("macos", "aarch64") => curr_dir_path.join("bin").join("bitcoind_macos"),
+            _ => curr_dir_path.join("bin").join("bitcoind"),
+        };
+        std::env::set_var(key, bitcoind_path);
+        let exe_path = bitcoind::exe_path().unwrap();
+
+        log::info!("Executable path: {:?}", exe_path);
+
+        let bitcoind = BitcoinD::with_conf(exe_path, &conf).unwrap();
+
+        // Generate initial 101 blocks
+        let mining_address = bitcoind
+            .client
+            .get_new_address(None, None)
+            .unwrap()
+            .require_network(bitcoind::bitcoincore_rpc::bitcoin::Network::Regtest)
+            .unwrap();
+        bitcoind
+            .client
+            .generate_to_address(101, &mining_address)
+            .unwrap();
+
+        // derive data directory
+        let data_dir = temp_dir.join("taker");
+
+        TakerCli { data_dir, bitcoind }
+    }
+
+    // Build a cli-command
+    fn execute(&self, cmd: &[&str]) -> String {
+        let mut args = Vec::new();
+
+        // Data directory
+        args.push("--data-directory");
+        args.push(&self.data_dir.as_os_str().to_str().unwrap());
+
+        // Full node network
+        args.push("--NETWORK");
+        args.push("regtest");
+
+        // Connection network type (clearnet/tor)
+        args.push("--network");
+        args.push("clearnet");
+
+        // RPC authentication (user:password) from the cookie file
+        //
+        // get rpc_auth
+        // Path to the cookie file
+        let cookie_file_path = Path::new(&self.bitcoind.params.cookie_file);
+
+        // Read the contents of the cookie file
+        let rpc_auth = fs::read_to_string(cookie_file_path).expect("failed to read from file");
+
+        args.push("--USER:PASSWORD");
+        args.push(&rpc_auth);
+
+        // Full node address for RPC connection
+        let rpc_address = self.bitcoind.params.rpc_socket.to_string();
+        args.push("--ADDRESS:PORT");
+        args.push(&rpc_address);
+
+        // Wallet name
+        args.push("--WALLET");
+        args.push("test_wallet");
+
+        // Custom arguments for the taker-cli command
+
+        // makers count
+        args.push("3");
+
+        // tx_count
+        args.push("3");
+
+        // fee_rate
+        args.push("1000");
```

Comment:
Good to keep these docs as we only will forget in 3 weeks which arg was for what.